src/Geometry/CubeGenerator.cpp

```diff
@@ -225,137 +237,118 @@ void seissol::geometry::CubeGenerator::cubeGenerator(unsigned int numCubes[4],
-
-  int ncVarElemNeighborRanks;
-  checkNcError(nc_def_var(
-      ncFile, "element_neighbor_ranks", NC_INT, 3, dimsElemSides, &ncVarElemNeighborRanks));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarElemNeighborRanks, NC_COLLECTIVE));
-
-  int ncVarElemMPIIndices;
-  checkNcError(
-      nc_def_var(ncFile, "element_mpi_indices", NC_INT, 3, dimsElemSides, &ncVarElemMPIIndices));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarElemMPIIndices, NC_COLLECTIVE));
-
-  int ncVarElemGroup;
-  checkNcError(nc_def_var(ncFile, "element_group", NC_INT, 2, dimsElemSides, &ncVarElemGroup));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarElemGroup, NC_COLLECTIVE));
-
-  int ncVarVrtxSize;
-  checkNcError(nc_def_var(ncFile, "vertex_size", NC_INT, 1, &ncDimPart, &ncVarVrtxSize));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarVrtxSize, NC_COLLECTIVE));
-
-  int ncVarVrtxCoords;
-  int dimsVrtxCoords[] = {ncDimPart, ncDimVrtx, ncDimDimension};
-  checkNcError(
-      nc_def_var(ncFile, "vertex_coordinates", NC_DOUBLE, 3, dimsVrtxCoords, &ncVarVrtxCoords));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarVrtxCoords, NC_COLLECTIVE));
-
-  int ncVarBndSize;
-  checkNcError(nc_def_var(ncFile, "boundary_size", NC_INT, 1, &ncDimPart, &ncVarBndSize));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarBndSize, NC_COLLECTIVE));
-
-  int ncVarBndElemSize;
-  int dimsBndElemSize[] = {ncDimPart, ncDimBnd};
-  checkNcError(
-      nc_def_var(ncFile, "boundary_element_size", NC_INT, 2, dimsBndElemSize, &ncVarBndElemSize));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarBndElemSize, NC_COLLECTIVE));
-
-  int ncVarBndElemRank;
-  checkNcError(
-      nc_def_var(ncFile, "boundary_element_rank", NC_INT, 2, dimsBndElemSize, &ncVarBndElemRank));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarBndElemRank, NC_COLLECTIVE));
-
-  int ncVarBndElemLocalIds;
-  int dimsBndElemLocalIds[] = {ncDimPart, ncDimBnd, ncDimBndElem};
-  checkNcError(nc_def_var(
-      ncFile, "boundary_element_localids", NC_INT, 3, dimsBndElemLocalIds, &ncVarBndElemLocalIds));
-  //      checkNcError(nc_var_par_access(ncFile, ncVarBndElemLocalIds, NC_COLLECTIVE));
+  int masterRank;
+  unsigned int groupSize = 1;
+#ifdef USE_MPI
+  groupSize = utils::Env::get<unsigned int>("SEISSOL_NETCDF_GROUP_SIZE", 1);
+  if (nProcs % groupSize != 0)
+    logError() << "#Processes must be a multiple of the group size" << groupSize;
+
+  int master = (rank / groupSize) * groupSize;
+  MPI_Comm commMaster;
+  MPI_Comm_split(
+      seissol::MPI::mpi.comm(), rank % groupSize == 0 ? 1 : MPI_UNDEFINED, rank, &commMaster);
+
+  masterRank = -1;
+
+  if (commMaster != MPI_COMM_NULL) {
+    MPI_Comm_rank(commMaster, &masterRank);
+  }
+#else  // USE_MPI
+  masterRank = rank; // = 0;
+#endif // USE_MPI
+  size_t bndSize = -1;
+  size_t bndElemSize = -1;
+
+  bool hasGroup = false;
+
+  int* sizes = 0L;
+  int maxSize = 0;
+  if (masterRank == 0) {
```

Comment:
Is that code still rank-dependent? Otherwise, the `if` can be removed, and the `MPI_Bcast` in line 296 as well.